src/analysis/processing/qgsalgorithmbasicstatistics.cpp

```diff
@@ -0,0 +1,421 @@
+    fields.append( QgsField( QStringLiteral( "mean" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "median" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "stddev" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "cv" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "minority" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "majority" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "q1" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "q3" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "iqr" ), QMetaType::Double ) );
+  }
+  else if ( field.isDateOrTime() )
+  {
+    if ( field.type() == QMetaType::Type::QDate )
+    {
+      fields.append( QgsField( QStringLiteral( "min" ), QMetaType::QDate ) );
+      fields.append( QgsField( QStringLiteral( "max" ), QMetaType::QDate ) );
+    }
+    else if ( field.type() == QMetaType::Type::QTime )
+    {
+      fields.append( QgsField( QStringLiteral( "min" ), QMetaType::QTime ) );
+      fields.append( QgsField( QStringLiteral( "max" ), QMetaType::QTime ) );
+    }
+    else
+    {
+      fields.append( QgsField( QStringLiteral( "min" ), QMetaType::QDateTime ) );
+      fields.append( QgsField( QStringLiteral( "max" ), QMetaType::QDateTime ) );
+    }
+    fields.append( QgsField( QStringLiteral( "range" ), QMetaType::Double ) );
+  }
+  else
+  {
+    fields.append( QgsField( QStringLiteral( "min" ), QMetaType::QString ) );
+    fields.append( QgsField( QStringLiteral( "max" ), QMetaType::QString ) );
+    fields.append( QgsField( QStringLiteral( "min_length" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "max_length" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "mean_length" ), QMetaType::Double ) );
+    fields.append( QgsField( QStringLiteral( "minority" ), QMetaType::QString ) );
+    fields.append( QgsField( QStringLiteral( "majority" ), QMetaType::QString ) );
+  }
+
+  QString destId;
+  std::unique_ptr< QgsFeatureSink > sink( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, destId, fields, Qgis::WkbType::NoGeometry, QgsCoordinateReferenceSystem() ) );
+  if ( parameters.value( QStringLiteral( "OUTPUT" ) ).isValid() && !sink )
+    throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+
+  QStringList data;
+  data << QObject::tr( "Analyzed field: %1" ).arg( fieldName );
+
+  QVariantMap outputs;
+
+  if ( field.isNumeric() )
+  {
+    outputs = calculateNumericStatistics( field, features, count, sink.get(), data, feedback );
+  }
+  else if ( field.isDateOrTime() )
+  {
+    outputs = calculateDateTimeStatistics( field, features, count, sink.get(), data, feedback );
+  }
+  else
+  {
+    outputs = calculateStringStatistics( field, features, count, sink.get(), data, feedback );
```

Comment:
We should pass the field index to these functions instead of the field name, and then access the attribute by field index instead. That'll be faster vs the multiple name-based lookups.

Reply:
Done. I still pass field to the `calculateDateTimeStatistics` method to get nice representation of datetime.